clients/admin-ui/cypress/e2e/systems-plus.cy.ts

```diff
@@ -156,6 +156,81 @@ describe("System management with Plus features", () => {
     });
   });
 
+  describe("data use", () => {
+    it("should enable legal basis editing if flexible is true", () => {
+      cy.fixture("systems/system_with_flexible_legal_basis.json").then(
+        (system) => {
+          cy.intercept("GET", "/api/v1/system/*", {
+            body: system,
+          }).as("getSystemWithFlexibleDataUses");
+        }
+      );
+      cy.visit(`${SYSTEM_ROUTE}/configure/flexible_system`);
+      cy.wait("@getSystemWithFlexibleDataUses");
+
+      cy.getByTestId("tab-Data uses").click();
+      cy.contains("a", "Analytics for Advertising Performance").click();
+
+      cy.contains("label", "This legal basis is flexible").should("exist");
+      cy.get('input[name="flexible_legal_basis_for_processing"]')
+        .should("exist")
+        .and("be.disabled");
+
+      cy.getByTestId("input-legal_basis_for_processing")
+        .get("input")
+        .should("be.enabled");
+
+      cy.getByTestId("save-btn").should("exist");
+    });
+
+    it("should disable legal basis editing if flexible is false", () => {
+      cy.fixture("systems/system_with_flexible_legal_basis.json").then(
+        (system) => {
+          cy.intercept("GET", "/api/v1/system/*", {
+            body: system,
+          }).as("getSystemWithFlexibleDataUses");
+        }
+      );
```

Comment:
same simplification as above can be done here 👍 . generally, `cy.intercept` with a `fixture` argument is the simplest way to do things. `cy.fixture()` with a nested `cy.intercept` is needed when you want to modify the fixture a little first. but overall, thanks for simplifying the fixture already!